modules/javafx.graphics/src/main/java/com/sun/javafx/application/PlatformImpl.java

```diff
@@ -1044,4 +1045,57 @@ private static boolean isSupportedImpl(ConditionalFeature feature) {
                 return Toolkit.getToolkit().isSupported(feature);
         }
     }
+
+    private static PlatformPreferences platformPreferences;
+
+    public static PlatformPreferences getPlatformPreferences() {
+        if (platformPreferences == null) {
+            throw new IllegalStateException("Toolkit not initialized");
+        }
+
+        return platformPreferences;
+    }
+
+    /**
+     * Called by Glass when the toolkit is initialized.
+     *
+     * @param wellKnownKeys a map of platform-specific keys to well-known keys
+     * @param preferences the initial set of platform preferences
+     */
+    public static void initPreferences(Map<String, String> wellKnownKeys, Map<String, Object> preferences) {
+        platformPreferences = new PlatformPreferences(wellKnownKeys);
+        platformPreferences.update(preferences);
+    }
+
+    /**
+     * Called by Glass when one or several platform preferences have changed.
+     * <p>
+     * This method can be called on any thread. The supplied {@code preferences} map may
+     * include all platform preferences, or only the changed preferences. If a preference
+     * was removed, the corresponding key is mapped to {@code null}.
+     *
+     * @param preferences a map that includes the changed preferences
+     */
+    public static void updatePreferences(Map<String, Object> preferences) {
+        if (isFxApplicationThread()) {
+            checkHighContrastThemeChanged(preferences);
+            platformPreferences.update(preferences);
+        } else {
+            // Make a defensive copy in case the caller of this method decides to re-use or
+            // modify its preferences map after the method returns. Don't use Map.copyOf
+            // because the preferences map may contain null values.
+            Map<String, Object> preferencesCopy = new HashMap<>(preferences);
+            runLater(() -> updatePreferences(preferencesCopy));
+        }
+    }
+
+    // This method will be removed when StyleThemes are added.
+    private static void checkHighContrastThemeChanged(Map<String, Object> preferences) {
+        if (preferences.get("Windows.SPI.HighContrastOn") == Boolean.TRUE) {
```

Comment:
This might not be an issue, considering that the Boolean(String) constructor is terminally deprecated anyway.

Reply:
It might not be, or it may become a bigger issue than it is now.  Why take the risk?  The rule in Java is always the same, objects are compared with `equals` unless you specifically need reference equality.

You don't do this with other primitive wrappers either (because you are probably aware of things like the integer cache, which is just an implementation detail and no guarantee).  Same for these booleans, you're relying on implementation details for this to work correctly.

Reply:
I agree with John: `if (Boolean.TRUE.equals(` is much better (maintaintability, correctness)